### PR TITLE
chore: adding sql question regarding distinct and count

### PR DESCRIPTION
**Pull request by ht-l1**
## Summary of changes

chore: adding sql question regarding distinct and count

## Checklist

- [x] I have read and followed the [contribution guidelines](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CONTRIBUTING.md).
- [x] I have read through the [Code of Conduct](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CODE_OF_CONDUCT.md) and agree to abide by the rules.
- [x] This PR is for one of the available issues and is not a PR for an issue already assigned to someone else.
- [x] My PR title has a short descriptive name so the maintainers can get an idea of what the PR is about.
- [x] I have provided a summary of my changes.

closes #XXXXX

### CODE REVIEW

File: src/data/sql-quiz.ts [hunk 1]
```diff
@@ -1033,6 +1033,19 @@ const sqlQuiz = [
...
+  },
+  {
+    Question:
+      "Which of the following SQL queries correctly shows the number of unique subjects taken by a student?",
+    Answer: "COUNT(DISTINCT SUBJECT_ID)",
+    Distractor1:
+      "COUNT(UNIQUE SUBJECT_ID)",
+    Distractor2:
+      "SUM(DISTINCT SUBJECT_ID)",
+    Distractor3: "DISTINCT COUNT(SUBJECT_ID)",
```
Comment (jdwilkin4) [inline src/data/sql-quiz.ts:1045]: all of the answers should be complete queries with the SELECT keyword included

File: src/data/sql-quiz.ts [hunk 2]
```diff
@@ -1033,6 +1033,19 @@ const sqlQuiz = [
...
+    Explanation:
+      "The SQL function COUNT(DISTINCT SUBJECT_ID) is used to find the number of unique values in the SUBJECT_ID column, which represents the number of unique subjects taken by a student.",
```
Comment (jdwilkin4) [inline src/data/sql-quiz.ts:1047]: the explanation should also have the complete query that includes the SELECT keyword

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Please review the requested changes.
Once those changes have been implemented, then we can merge your PR into main. 👍

File: src/data/sql-quiz.ts (hunk 1 above)
Comment (ht-l1) [reply src/data/sql-quiz.ts]: just made the update to include the SELECT keyword

File: src/data/sql-quiz.ts (hunk 2 above)
Comment (ht-l1) [reply src/data/sql-quiz.ts]: just made the update to include the SELECT keyword

File: src/data/sql-quiz.ts [hunk 3]
```diff
@@ -1033,6 +1033,19 @@ const sqlQuiz = [
     Explanation:
       "To select some particular date between one date and another, you can use the BETWEEN and AND keywords while specifying the dates. You can also use comparison operators such as greater than (>), less than (<), or equal(=).",
     Link: "https://www.freecodecamp.org/news/sql-date-function-query-timestamp-example-format/"
+  },
+  {
+    Question:
+      "Which of the following SQL queries correctly shows the number of unique subjects taken by a student?",
+    Answer: "SELECT COUNT(DISTINCT SUBJECT_ID)",
```
Comment (jdwilkin4) [inline src/data/sql-quiz.ts:1040]: I am not a SQL expert, but this looks like it is missing the from clause.

This should be valid

```suggestion
    Answer: "SELECT COUNT(DISTINCT SUBJECT_ID) FROM table_name;",
```

File: src/data/sql-quiz.ts [hunk 4]
```diff
@@ -1033,6 +1033,19 @@ const sqlQuiz = [
...
+    Distractor1:
+      "SELECT COUNT(UNIQUE SUBJECT_ID)",
+    Distractor2:
+      "SELECT SUM(DISTINCT SUBJECT_ID)",
+    Distractor3: "SELECT DISTINCT COUNT(SUBJECT_ID)",
+    Explanation:
+      "The SQL function SELECT COUNT(DISTINCT SUBJECT_ID) is used to find the number of unique values in the SUBJECT_ID column, which represents the number of unique subjects taken by a student.",
```
Comment (jdwilkin4) [inline src/data/sql-quiz.ts:1047]:
```suggestion
      "The SQL function SELECT COUNT(DISTINCT SUBJECT_ID) FROM table_name; is used to find the number of unique values in the SUBJECT_ID column, which represents the number of unique subjects taken by a student.",
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
two last comments to make sure we have valid queries and then we should be good to go

**Review by jdwilkin4 — APPROVED** [review]
Congrats on another contribution to our repository. 🎉
We look forward to future contributions. 👍

**Pull request merged**